src/software/ai/navigator/path_planner/path_planner_test.cpp

```diff
@@ -61,7 +61,7 @@ std::vector<PlannerTestCase>
                   {.name  = "Circle of robots surrounding friendly robot at distance 3",
                    .start = Point(0, 0),
                    .end   = Point(4, 0),
-                   .navigable_area = Rectangle({-5, -5}, {5, 5}),
+                   .navigable_area = Rectangle({-4.5, -4.5}, {4.5, 4.5}),
```

Comment:
Before this change, the test would fail because of misalignment of the grid with the start point.

Reply:
What exactly do you mean? With the larger area, even if the start point didn't align with the grid it shouldn't have a problem finding the nearest one and going from there? It's kinda odd that making the area smaller makes it work.

Reply:
With `.navigable_area = Rectangle({-5, -5}, {5, 5})`, the start point is misaligned from the grid by 0.05m, so ThetaStar snaps that point to a point 0.05m off. Then it uses that new point when checking line of sight and finds no intersections. In the final step, ThetaStar replaces the start point with the real start point and that causes the first point in the path and the second point in the path to intersect the obstacle. I wish there were an easy way to fix this, but it seems part of the tradeoff of using a grid

Reply:
Oh. What about appending `(real_start -> grid_start)` at the front of the path, rather than snapping, doing the calculation, then using the original?

Reply:
Since real_start is never exactly equal to grid_start and the MovePrimitive (as it is right now) always goes to the second point in the path, that idea would basically mean we'll oscillate around the start forever

Reply:
Hmmm ok I can't really think of a solution here.